### PR TITLE
Replace new instantiation of BuildInfo with calls to Injector

### PR DESCRIPTION
**Pull request by calixtus**
closes https://github.com/JabRef/jabref-issue-melting-pot/issues/604

also removes some telemetry artifacts

### Mandatory checks

- [ ] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [ ] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [ ] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

### CODE REVIEW

**Review by koppor — DISMISSED** [review]
Looks good - you decided against constructor-based injection, but sued the `Injector` directly? If it is OK for you, it is OK for me!

**Comment by calixtus** [comment]
I have no hard opinion on this atm. Probably think about some hours. Will come back to this later

**Comment by koppor** [comment]
I think, constructor-based injection will increase the code, but the dependencies are explicit. In constract, with the injector, one "just" needs to search for class usages with an Injector... 

Using the example of `BuildInfo` can guide us to fix an `ADR`. -- I mean: Implement with `Injector`  (done here), implement as constructor passing (maybe in another PR) and then compare.

**Comment by Siedlerchr** [comment]
In general, constructor based dependency injection is preferable because you can better test it and it's more explicit. For example, spring also recommends ctor based injection

File: src/main/java/org/jabref/gui/entryeditor/citationrelationtab/semanticscholar/SemanticScholarFetcher.java [hunk 1]
```diff
@@ -86,10 +89,10 @@ public List<BibEntry> searchCiting(BibEntry entry) throws FetcherException {
 
     @Override
     public String getName() {
-        return "Semantic Scholar Citations Fetcher";
+        return FETCHER_NAME;
     }
 
     private String getApiKey() {
-        return importerPreferences.getApiKey(getName()).orElse(API_KEY);
+        return importerPreferences.getApiKey(getName()).orElse(apiKey);
```
Comment (koppor) [inline src/main/java/org/jabref/gui/entryeditor/citationrelationtab/semanticscholar/SemanticScholarFetcher.java:96]: The fallback is wrong; the default key is now in the ImporterPreferences? Here, `""``  should be returned?

File: src/main/java/org/jabref/logic/preferences/JabRefCliPreferences.java [hunk 2]
```diff
@@ -2085,6 +2091,22 @@ private List<String> getFetcherKeysFromKeyring(List<String> names) {
         return keys;
     }
 
+    private Map<String, String> getDefaultFetcherKeys() {
+        BuildInfo buildInfo = Injector.instantiateModelOrService(BuildInfo.class);
+        if (buildInfo == null) {
+            return Collections.emptyMap();
```
Comment (koppor) [inline src/main/java/org/jabref/logic/preferences/JabRefCliPreferences.java:2097]: Add `LOGGER.warn` here.

File: docs/code-howtos/fetchers.md [hunk 3]
```diff
@@ -79,7 +79,7 @@ In `build.gradle`, these variables are filled:
 The `BuildInfo` class reads from that file.
 
 ```java
-new BuildInfo().springerNatureAPIKey
+Injector.instantiateModelOrService(BuildInfo.class).springerNatureAPIKey
```
Comment (koppor) [inline docs/code-howtos/fetchers.md:82]: This is wrong now - it should go through the ImporterPreferences.

**Review by koppor — CHANGES_REQUESTED** [review]
I like keeping the preferences together. - Some code still needs to be adapted to use 

    importerPreferences.getApiKey(getName())

instead of

    Injector.instantiateModelOrService(BuildInfo.class).semanticScholarApiKey

because the keys are now provided by importerPreferences.getApiKey

**Comment by InAnYan** [comment]
https://github.com/JabRef/jabref/pull/11954#issuecomment-2410311010

Then what is the purpose of DI framework? Like the one we use?

**Comment by calixtus** [comment]
Afterburner is not a di framework.
It just provides some basic functionality to provide injections into presenters.
See discussion here: https://github.com/JabRef/jabref/pull/11342#discussion_r1631031758

**Review by koppor — APPROVED** [review]

**Pull request merged**